extensions/jackson/runtime/src/main/java/io/quarkus/jackson/runtime/HybridJacksonPool.java

```diff
@@ -0,0 +1,195 @@
+            if (stripesCount <= 0) {
+                throw new IllegalArgumentException("Expecting a stripesCount that is larger than 0");
+            }
+
+            int size = roundToPowerOfTwo(stripesCount);
+            this.heads = new AtomicReferenceArray<>(size * CACHE_LINE_PADDING);
+
+            int mask = (size - 1) << CACHE_LINE_SHIFT;
+            this.threadProbe = new XorShiftThreadProbe(mask);
+        }
+
+        @Override
+        public BufferRecycler acquirePooled() {
+            int index = threadProbe.index();
+
+            Node currentHead = heads.get(index);
+            while (true) {
+                if (currentHead == null) {
+                    return new VThreadBufferRecycler(index);
+                }
+
+                Node witness = heads.compareAndExchange(index, currentHead, currentHead.next);
+                if (witness == currentHead) {
+                    currentHead.next = null;
+                    return currentHead.value;
+                } else {
+                    currentHead = witness;
+                }
+            }
+        }
+
+        @Override
+        public void releasePooled(BufferRecycler recycler) {
+            VThreadBufferRecycler vThreadBufferRecycler = (VThreadBufferRecycler) recycler;
+            Node newHead = new Node(vThreadBufferRecycler);
+
+            Node next = heads.get(vThreadBufferRecycler.slot);
+            while (true) {
+                Node witness = heads.compareAndExchange(vThreadBufferRecycler.slot, next, newHead);
+                if (witness == next) {
+                    newHead.next = next;
+                    return;
+                } else {
+                    next = witness;
+                }
+            }
+        }
+
+        private static class Node {
+            final VThreadBufferRecycler value;
+            Node next;
+
+            Node(VThreadBufferRecycler value) {
+                this.value = value;
+            }
+        }
+    }
+
+    private static class VThreadBufferRecycler extends BufferRecycler {
+        private final int slot;
+
+        VThreadBufferRecycler(int slot) {
+            this.slot = slot;
+        }
+    }
+
+    private static class VirtualPredicate {
+        private static final MethodHandle virtualMh = findVirtualMH();
+
+        private static MethodHandle findVirtualMH() {
+            try {
+                return MethodHandles.publicLookup().findVirtual(Thread.class, "isVirtual", MethodType.methodType(boolean.class));
+            } catch (Exception e) {
+                return null;
+            }
+        }
+
+        private static Predicate<Thread> findIsVirtualPredicate() {
+            return virtualMh != null ? t -> {
+                try {
+                    return (boolean) virtualMh.invokeExact(t);
+                } catch (Throwable e) {
+                    throw new RuntimeException(e);
+                }
+            } : t -> false;
```

Comment:
We reallly try hard to avoid using lambas as much as possible in code that ends up on the runtime classpath

Reply:
Thanks for reminder, fixed.

Reply:
🙏